File: python/scripts/list_resource_types.py
import os
from travelperk_http_python.builder.builder import build
from utils import write_output
travelperk = build(os.environ["API_KEY"], False)
resource_types = travelperk.scim().discovery().resource_types()
write_output("list_resource_types", [
    resource_types["totalResults"],
    resource_types["itemsPerPage"],
    resource_types["startIndex"],
    resource_types["schemas"][0],
    resource_types["Resources"][0]["schemas"][0],
    resource_types["Resources"][0]["id"],
    resource_types["Resources"][0]["name"],
    resource_types["Resources"][0]["endpoint"],
    resource_types["Resources"][0]["description"],
    resource_types["Resources"][0]["schema"],
    resource_types["Resources"][0]["schemaExtensions"][0]["schema"],
    resource_types["Resources"][0]["schemaExtensions"][0]["required"],
    resource_types["Resources"][0]["schemaExtensions"][1]["schema"],
    resource_types["Resources"][0]["schemaExtensions"][1]["required"],
    resource_types["Resources"][0]["meta"]["resourceType"],
    resource_types["Resources"][0]["meta"]["location"],
])

File: php/scripts/list_resource_types.php
<?php
require_once(__DIR__."/../vendor/autoload.php");
require_once("utils.php");
use Namelivia\TravelPerk\ServiceProvider;
$isSandbox = false;
$travelperk = (new ServiceProvider())->build(getenv("API_KEY"), false);
$resourceTypes = $travelperk->scim()->discovery()->resourceTypes();
write_output("list_resource_types", [
	$resourceTypes->totalResults,
	$resourceTypes->itemsPerPage,
	$resourceTypes->startIndex,
	$resourceTypes->schemas[0],
	$resourceTypes->Resources[0]->schemas[0],
	$resourceTypes->Resources[0]->id,
	$resourceTypes->Resources[0]->name,
	$resourceTypes->Resources[0]->endpoint,
	$resourceTypes->Resources[0]->description,
	$resourceTypes->Resources[0]->schema,
	$resourceTypes->Resources[0]->schemaExtensions[0]->schema,
	$resourceTypes->Resources[0]->schemaExtensions[0]->required,
	$resourceTypes->Resources[0]->schemaExtensions[1]->schema,
	$resourceTypes->Resources[0]->schemaExtensions[1]->required,
	$resourceTypes->Resources[0]->meta->resourceType,
	$resourceTypes->Resources[0]->meta->location,
]);
